sql/parse/parse.go

```diff
@@ -35,6 +35,7 @@ var (
 	showIndexRegex       = regexp.MustCompile(`^show\s+(index|indexes|keys)\s+(from|in)\s+\S+\s*`)
 	describeRegex        = regexp.MustCompile(`^(describe|desc|explain)\s+(.*)\s+`)
 	fullProcessListRegex = regexp.MustCompile(`^show\s+(full\s+)?processlist$`)
+	setAutocommitRegexp  = regexp.MustCompile(`^set\s+(@+)session.autocommit\s*=`)
```

Comment:
there is already support for `set` in go-mysql-server, there is no need to add a regexp for this

Reply:
Yes, but because ON is a keyword, yacc Parse fails. That's why I first converts on/off to 1/0 and pass through regular workflow.

Reply:
Then maybe we can pre-process the query and replace the on/off with 1/0 before actually parsing it using sqlparser. There might be other queries that use on/off, this is very ad-hoc.